crates/telio-traversal/src/session_keeper.rs

```diff
@@ -194,113 +170,81 @@ impl SessionKeeperTrait for SessionKeeper {
-                            if let Err(e) = ping(&c.pingers, (&public_key, &dual_target)).await {
-                                telio_log_warn!(
-                                    "Failed to batch-ping, peer with key: {:?}, error: {:?}",
-                                    public_key,
-                                    e
-                                );
-                            }
-                            Ok(())
-                        })
-                    }),
-                );
-
-                Ok(())
-            })
-            .await
-            .map_err(Error::Task)?,
 
-            None => task_exec!(&self.task, async move |s| {
-                if s.nonbatched_actions.contains_action(&public_key) {
-                    let _ = s.nonbatched_actions.remove_action(&public_key);
-                }
+        let batch_all = self.batch_all;
+        telio_log_debug!(
+            "Add action for {} and interval {:?}. batch_all({})",
+            public_key,
+            interval,
+            batch_all
+        );
 
-                Ok(s.nonbatched_actions.add_action(
-                    public_key,
-                    interval,
-                    Arc::new(move |c| {
-                        Box::pin(async move {
-                            if let Err(e) = ping(&c.pingers, (&public_key, &dual_target)).await {
-                                telio_log_warn!(
-                                    "Failed to ping, peer with key: {:?}, error: {:?}",
-                                    public_key,
-                                    e
-                                );
-                            }
-                            Ok(())
-                        })
-                    }),
-                ))
-            })
-            .await
-            .map_err(Error::Task)?
-            .map_err(Error::RepeatedActionError)
-            .map(|_| ())?,
-        }
+        task_exec!(&self.task, async move |s| {
+            if s.actions.contains_action(&public_key) {
+                let _ = s.actions.remove_action(&public_key);
+            }
 
+            if batch_all {
+                s.actions.set_all_immediate();
+            }
+
+            Ok(s.actions.add_action(
+                public_key,
+                interval,
+                Arc::new(move |c| {
+                    Box::pin(async move {
+                        if let Err(e) = ping(&c.pingers, (&public_key, &dual_target)).await {
+                            telio_log_warn!(
+                                "Failed to ping, peer with key: {:?}, error: {:?}",
+                                public_key,
+                                e
+                            );
+                        }
+                        Ok(())
+                    })
+                }),
+            ))
+        })
+        .await
+        .map_err(Error::Task)?
+        .map_err(Error::RepeatedActionError)
+        .map(|_| ())?;
```

Comment:
`.map(|_| ())?;` I think this is a no-op and can be removed

Reply:
You're right, removed it. Probably previously it was inlined directly into `Ok()` so it made sense